lambdas/data-migration1/tests/test-rules.js

```diff
@@ -0,0 +1,394 @@
+    omit(
+      {
+        ...fakeRule,
+        arn: fakeRule.rule.arn ? fakeRule.rule.arn : null,
+        type: fakeRule.rule.type,
+        value: fakeRule.rule.value ? fakeRule.rule.value : null,
+        enabled: true,
+        log_event_arn: null,
+        execution_name_prefix: null,
+        payload: null,
+        queue_url: null,
+        tags: null,
+        created_at: new Date(fakeRule.createdAt),
+        updated_at: new Date(fakeRule.updatedAt),
+      },
+      ['createdAt', 'updatedAt', 'state', 'provider', 'collection', 'rule']
+    )
+  );
+});
+
+test.serial('migrateRuleRecord throws error on invalid source data from DynamoDb', async (t) => {
+  fakeCollection = fakeCollectionFactory();
+  fakeProvider = fakeProviderFactory({
+    encrypted: false,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  });
+  const fakeRule = generateFakeRule(fakeCollection.name, fakeCollection.version, fakeProvider.id);
+
+  // make source record invalid
+  delete fakeRule.files;
+
+  await t.throwsAsync(migrateRuleRecord(fakeRule, t.context.knex));
+});
+
+test.serial('migrateRuleRecord handles nullable fields on source rule data', async (t) => {
+  fakeCollection = fakeCollectionFactory();
+  fakeProvider = fakeProviderFactory({
+    encrypted: false,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  });
+  const { knex, providerKmsKeyId } = t.context;
+  const fakeRule = generateFakeRule(fakeCollection.name, fakeCollection.version, fakeProvider.id);
+
+  delete fakeRule.rule.value;
+  delete fakeRule.rule.arn;
+  delete fakeRule.payload;
+  delete fakeRule.logEventArn;
+  delete fakeRule.queueUrl;
+  delete fakeRule.meta;
+  delete fakeRule.tags;
+
+  await migrateCollectionRecord(fakeCollection, knex);
+  await migrateProviderRecord(fakeProvider, providerKmsKeyId, knex);
+  await migrateRuleRecord(fakeRule, t.context.knex);
+  const createdRecord = await t.context.knex.queryBuilder()
+    .select()
+    .table('rules')
+    .where({ name: fakeRule.name })
+    .first();
+
+  t.deepEqual(
+    omit(createdRecord, ['cumulus_id', 'collection_cumulus_id', 'provider_cumulus_id']),
+    omit(
+      {
+        ...fakeRule,
+        arn: fakeRule.rule.arn ? fakeRule.rule.arn : null,
+        value: fakeRule.rule.value ? fakeRule.rule.value : null,
+        type: fakeRule.rule.type,
+        enabled: true,
+        log_event_arn: null,
+        execution_name_prefix: null,
+        payload: null,
+        queue_url: null,
+        meta: null,
+        tags: null,
+        created_at: new Date(fakeRule.createdAt),
+        updated_at: new Date(fakeRule.updatedAt),
+      },
+      ['createdAt', 'updatedAt', 'state', 'provider', 'collection', 'rule']
+    )
+  );
+});
+
+test.serial('migrateRuleRecord ignores extraneous fields from Dynamo', async (t) => {
```

Comment:
Does `rule` count? Because we have in the DynamoDB schema 
```
rule: {
      title: 'Ingest Rule',
      type: 'object',
      properties: {
        type: {
          type: 'string',
          enum: ['onetime', 'scheduled', 'sns', 'kinesis', 'sqs'],
        },
        // Value is multi-use.   For a kinesis rule this is the target stream arn, for
        // a scheduled event it's the schedule pattern (e.g. cron), for a one-time rule.
        value: {
          type: 'string',
        },
        // Kinesis scheduled event arn
        arn: {
          type: 'string',
          readonly: true,
        },
        // Kinesis scheduled log event arn
        logEventArn: {
          type: 'string',
          readonly: true,
        },
      },
      required: ['type'],
    },
```

but we don't maintain this structure

Reply:
Good point. Though I think this case is sufficiently covered by this test?

https://github.com/nasa/cumulus/pull/2000/files#diff-d7b0058e0252e8083289a57a228dc6cec19ad3877e20ff8c22626a419ae27affR160

If there was a `rule` property returned on the `createdRecord` in that test, I think it would fail. But feel free to verify that

If that is true, then this test isn't necessary

Reply:
Yeah, after looking at it more, I'm pretty sure that this test is unnecessary

Reply:
I see, that's a good point. I agree and will remove the test!